packages/block-editor/src/components/link-control/index.js

```diff
@@ -350,15 +348,57 @@ function LinkControl( {
 				/>
 			) }
 
-			<div className="block-editor-link-control__drawer">
-				{ showSettingsDrawer && (
-					<div className="block-editor-link-control__tools">
-						<LinkControlSettingsDrawer
-							value={ value }
-							settings={ settings }
-							onChange={ onChange }
+			<div className="block-editor-link-control__tools">
+				{ ( showSettings || showTextControl ) && (
+					<>
+						<Button
+							aria-expanded={ settingsOpen }
+							onClick={ () => setSettingsOpen( ! settingsOpen ) }
+							icon={ settingsIcon }
+							label={ __( 'Toggle link settings' ) }
+							aria-controls="link-1" // todo - this should be dynamic
 						/>
-					</div>
+						<AnimatePresence>
+							{ settingsOpen && (
+								<motion.div
+									className="block-editor-link-control__drawer"
+									hidden={ ! settingsOpen }
+									id={ 'link-1' } // todo - this should be dynamic
+									initial="collapsed"
+									animate="open"
+									exit="collapsed"
+									variants={ {
+										open: { opacity: 1, height: 'auto' },
+										collapsed: { opacity: 0, height: 0 },
+									} }
+									transition={ {
+										duration: 0.2,
```

Comment:
This still feels too slow to me

Reply:
Ok no problem.

Do we not have a standard animation duration? Anyway, happy for anyone to push a commit with an alternative duration.

Reply:
I changed it to .1 which feels better to me. I don't see a standard - I think it depends on the interaction.